### PR TITLE
fix: enum values, hasChanged checks and optional attributes

### PR DESCRIPTION
**Pull request by stebenz**
…values

### CODE REVIEW

File: zitadel/v2/application_oidc/funcs.go [hunk 1]
```diff
@@ -53,81 +53,95 @@ func update(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
...
+			!reflect.DeepEqual(respTypes, oidcConfig.GetResponseTypes()) ||
+			!reflect.DeepEqual(grantTypes, oidcConfig.GetGrantTypes()) ||
+			appType != oidcConfig.AppType.String() ||
+			authMethodType != oidcConfig.AuthMethodType.String() ||
+			!reflect.DeepEqual(postLogoutRedirectURIs, oidcConfig.GetPostLogoutRedirectUris()) ||
+			devMode != oidcConfig.DevMode ||
+			accessTokenType != oidcConfig.AccessTokenType.String() ||
+			accessTokenRoleAssertion != oidcConfig.AccessTokenRoleAssertion ||
+			clockSkew.String() != oidcConfig.ClockSkew.String() ||
+			!reflect.DeepEqual(additionalOrigins, oidcConfig.GetAdditionalOrigins()) {
```
Comment (eliobischof) [inline zitadel/v2/application_oidc/funcs.go:124]: Why do we also need to compare to the system state? Isn't the comparision to the terraform state enough?

File: zitadel/v2/human_user/funcs.go [hunk 2]
```diff
@@ -145,67 +118,75 @@ func update(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
...
-		})
-		if err != nil {
-			return diag.Errorf("failed to update human profile: %v", err)
+	if d.HasChanges(firstNameVar, lastNameVar, nickNameVar, displayNameVar, preferredLanguageVar, genderVar) {
+		nickname := d.Get(nickNameVar)
+		displayname := d.Get(displayNameVar)
+		prefLang := d.Get(preferredLanguageVar)
+		gender := d.Get(genderVar)
+
+		if currentHuman.GetProfile().GetFirstName() != d.Get(firstNameVar).(string) ||
```
Comment (eliobischof) [inline zitadel/v2/human_user/funcs.go:141]: same as above

File: zitadel/v2/human_user/resource.go [hunk 3]
```diff
@@ -77,16 +77,16 @@ func GetResource() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "Preferred language of the user",
-				Computed:    true,
+				Default:     defaultPreferredLanguage,
```
Comment (eliobischof) [inline zitadel/v2/human_user/resource.go:80]: doesn't ZITADEL set this value itself if it gets an empty string?

Comment (stebenz) [reply zitadel/v2/human_user/resource.go:80]: that's the problem, as in terraform an empty string can't be mapped and results in 0 which in turn gets mapped by ZITADEL to the default, so to handle it correctly in the state, the default is set if nothing is provided

Comment (eliobischof) [reply zitadel/v2/human_user/resource.go:80]: > results in 0 which in turn gets mapped by ZITADEL to the default,
isn't that exaxtly what we want? But ok, you know what you are doing.

**Review by eliobischof — APPROVED** [review]

**Pull request merged**